zerver/lib/markdown/__init__.py

```diff
@@ -2582,9 +2582,7 @@ def do_convert(
     if md_engine_key in md_engines:
         _md_engine = md_engines[md_engine_key]
     else:
-        if DEFAULT_MARKDOWN_KEY not in md_engines:
-            maybe_update_markdown_engines(linkifiers_key=None, email_gateway=False)
-
+        maybe_update_markdown_engines(linkifiers_key=None, email_gateway=False)
         _md_engine = md_engines[(DEFAULT_MARKDOWN_KEY, email_gateway)]
```

Comment:
This is probably some kind of bug, introduced in 65838bb82540559a29be009495a29bf49515c100 (#7335). It’s really unclear what the logic is supposed to be, but this preserves the existing behavior.

Reply:
There's a broader set of bugs here; I'm going to clean this up in a new commit.